service/common/src/main/java/org/apache/polaris/service/persistence/InMemoryPolarisMetaStoreManagerFactory.java

```diff
@@ -59,8 +58,9 @@ public InMemoryPolarisMetaStoreManagerFactory(
     this.storageIntegration = storageIntegration;
   }
 
-  public void onStartup(RealmContextConfiguration realmContextConfiguration) {
-    bootstrapRealmsAndPrintCredentials(realmContextConfiguration.realms());
+  @Override
+  public void initializeForService(@Nonnull List<String> realmIds, @Nonnull String defaultRealmId) {
+    bootstrapRealmsAndPrintCredentials(realmIds);
```

Comment:
It seems kinda problematic that we are peeling apart `RealmContextConfiguration` into some of its children here. What if a `MetaStoreManagerFactory` needs the content of some custom `RealmContextConfiguration` to bootstrap?

Reply:
The other only attributes of `RealmContextConfiguration` are HTTP/REST related. I don't think that it's a good idea to mix "higher level" persistence concerns with HTTP/REST things. Aka it's IMO wrong to let a persistence implementation rely on `:polaris-service-common`.

Reply:
`RealmContextConfiguration` is an interface

Reply:
... in a module that mustn't be used in polaris-quarkus-admin.

If you wanna do things with that interface somewhere, feel free to do that. But please lets not mix Polaris-service with Polaris-admin concerns in OSS.

Reply:
>  in a module that mustn't be used in polaris-quarkus-admin.

Must is a strong word. Why must it not be used there?

Reply:
Maybe I assumed that you know, but that module contains Polaris _service_ code, which _breaks_ the admin tool.

Reply:
If I understand you correctly, you're saying that you don't want to simply add `:polaris-service-common` as a dependency to `polaris-quarkus-admin`. If we just blindly do that, the build will break. That much makes sense.

However it's not clear to me why this interface necessarily must reside where it does, or why we can't restructure the modules so as to avoid having to break this existing public interface.

Reply:
@eric-maynard : the old `onStartup` method had the single purpose of bootstrapping the in-memory impl. automatically.

I do not think it having `RealmContextConfiguration` as a parameter was a deliberate design decision in the first place :)

Reply:
Yeah that makes sense, we can always rip it out and add it back in if it's needed somewhere